### PR TITLE
Handle signals even when user app is not up

### PR DESCRIPTION
**Pull request by LMWF**
# Description

The behavior is a bit different today from when the issue was first filed - a ctrl+c to the cli window does kill daprd even if the user's app is not up.
However sending SIGTERM or SIGKILL directly to daprd are ignored until the user app is up.

This fixes the problems and also makes daprd always call a "cleanup" function for any last work that must be done.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_875_

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [x] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation
* [ ] Specification has been updated
* [ ] Provided sample for the feature

### CODE REVIEW

File: cmd/daprd/main.go [hunk 1]
```diff
@@ -97,6 +97,15 @@ func main() {
 		log.Fatal(err)
 	}
 
+	stop := make(chan os.Signal, 1)
+	signal.Notify(stop, syscall.SIGTERM, os.Interrupt)
+
+	go func() {
+		// handle signals and perform any cleanup before exiting
+		cleanup(rt, stop)
```
Comment (yaron2) [inline cmd/daprd/main.go]: We should call `cleanup` once, calling the method `cleanup` in line `280` is misleading, as it's not a cleanup method in that context, its blocking and then might clean up..

Comment (yaron2) [reply cmd/daprd/main.go]: Look at how [Sentry](https://github.com/dapr/dapr/blob/c5ac112253968bb37bb89e2156e27487c98b73b7/pkg/sentry/sentry.go#L47) is dealing with it, we should probably do something similar.

**Comment by amanbha** [comment]
@LMWF is this PR still valid?

**Comment by amanbha** [comment]
@LMWF this PR is old and I am closing it for now to keep PRs manageable across all our repos. Feel free to reopen when you plan to work on it again.
Looking forward to your contribution, thanks!

**Pull request closed without merging**